acceptance_tests/features/steps/unaddressed_steps.py

```diff
@@ -27,6 +34,49 @@ def check_uac_message_is_received(context):
     assert context.expected_message_received
 
 
+@given("I have the QID and UAC pairs ready")
+def generate_qid_uac_pairs(context):
+    context.batch_id = uuid4()
+
+    context.config_file = Path(__file__).parents[3].joinpath('resources', 'unaddressed_batch',
+                                                             'unaddressed_batch.csv').resolve(strict=True)
+    generate_messages_from_config_file_path(context.config_file, context.batch_id)
+    context.message_count = 0
+    start_listening_to_rabbit_queue(Config.RABBITMQ_RH_OUTBOUND_UAC_QUEUE,
+                                    functools.partial(_stop_consuming_after_n_messages, context=context, n=30))
+
+
+@when("the print files are generated")
+def generate_print_files(context):
+    context.print_file_paths = generate_print_files_from_config_file_path(context.config_file, context.test_file_path,
+                                                                          context.batch_id)
+
+
+@then("the print files contain the correct data")
+def validate_print_file_data(context):
+    manifests = [file_path for file_path in context.print_file_paths if file_path.suffix == '.manifest']
+    print_files = [file_path for file_path in context.print_file_paths if file_path.suffix == '.csv']
+
+    assert len(manifests) == 3, 'Incorrect number of manifest files'
+
+    with open(context.config_file) as batch_config:
+        config_file = list(csv.DictReader(batch_config))
+
+    for index, print_file in enumerate(print_files):
+        with open(print_file) as print_file_handler:
+            print_file_reader = csv.DictReader(print_file_handler, delimiter='|', fieldnames=PRINT_FILE_TEMPLATE)
+
+            row_counter = 0
+            for row in print_file_reader:
```

Comment:
Could `enumerate` be used here as well?

```suggestion
            for count, row in enumerate(print_file_reader):
```

Reply:
Good shout. This file has now been moved to [here](https://github.com/ONSdigital/census-rm-qid-batch-runner/pull/5/files#diff-0b4c33d154623b732d19eea287755318), so I'll make the changes there 👍

Reply:
Thinking about it, I think we wanted to use the `row_counter` outside of the loop for checking the total row count against the expected batch count. I wonder if there's a nice way of getting the row count from DictReader instead